src/lib/GSENSOR/stk8baxx.cpp

```diff
@@ -0,0 +1,284 @@
+#ifdef HAS_GSENSOR_STK8xxx
+#include "stk8baxx.h"
+#include "logging.h"
+
+#define STK8xxx_SLAVE_ADDRESS	0x18
+
+#define	STK_REG_POWMODE		0x11
+#define	STK_SUSPEND_MODE	0x80
+
+#define STK8xxx_REG_RANGESEL	0x0F
+#define STK8xxx_RANGE_2G		0x03
+#define STK8xxx_RANGE_4G		0x05
+#define STK8xxx_RANGE_8G		0x08
+
+#define STK_REG_CHIPID		0x00
+/* STK8321 or STK8323 CHIP ID = 0x23 */
+#define STK8xxx_CHIPID_VAL  0x23
+#define STK8327_CHIPID_VAL  0x26
+/* STK8BAxx */
+ /* S or R resolution = 10 bit */
+#define STK8BA50_X_CHIPID_VAL	0x86
+#define STK8BA5X_CHIPID_VAL		0x87
+
+#define PID_SIZE	16
+uint8_t chipid_temp = 0x00;
+uint8_t stk8xxx_pid_list[PID_SIZE] = {STK8xxx_CHIPID_VAL, STK8BA50_X_CHIPID_VAL, STK8BA5X_CHIPID_VAL, STK8327_CHIPID_VAL};
+
+void STK8xxx::ReadAccRegister(uint8_t reg, uint8_t *data)
+{
+    Wire.beginTransmission(STK8xxx_SLAVE_ADDRESS);
+    Wire.write(reg);
+    Wire.endTransmission();
+
+    Wire.requestFrom(STK8xxx_SLAVE_ADDRESS, 1);    // request 1 bytes from slave device
+    *data = Wire.read(); // receive a byte
+}
+
+void STK8xxx::WriteAccRegister(uint8_t reg, uint8_t data)
+{
+    Wire.beginTransmission(STK8xxx_SLAVE_ADDRESS);
+    Wire.write(reg);
+    Wire.write(data);
+    Wire.endTransmission();
+}
+
+/*
+ * Anymotion init register
+ * Must be placed after the software reset
+ *
+ */
+void STK8xxx::STK8xxx_Anymotion_init()
+{
+	unsigned char ARegAddr, ARegWriteValue;
+	
+	/* Enable X Y Z-axis any-motion (slope) interrupt */
+    ARegAddr       = 0x16;
```

Comment:
Should all of the following ARegAddr and ARegWriteValue be defines in the header?

Reply:
Got it, fix it later